### PR TITLE
Feature/webumenia 1270

### PR DESCRIPTION
**Pull request by igor-kamil**
# Description

Simple Laravel Mix setup to compile `style.css`

Fixes [WEBUMENIA-1270](https://jira.sng.sk/browse/WEBUMENIA-1270)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist:

- [x] I have updated the [CHANGELOG](../CHANGELOG.md)
- [x] I have requested at least 1 reviewer for this PR
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,12 +1,24 @@
...
+    "devDependencies": {
+        "cross-env": "^5.1",
+        "jquery": "^3.4.1",
+        "laravel-mix": "^4.0.7",
+        "less": "^3.11.1",
+        "less-loader": "^5.0.0",
+        "resolve-url-loader": "^3.1.0",
+        "sass": "^1.15.2",
+        "sass-loader": "^7.1.0",
+        "vue-template-compiler": "^2.6.11"
```
[the remark this answers is not in the document]
Comment (igor-kamil) [reply package.json:22]: no. but it was loaded automatically after I hit `npm run watch` ..

File: public/css/less/bootstrap/glyphicons.less [hunk 2]
```diff
@@ -1,305 +0,0 @@
-//
```
[the remark this answers is not in the document]
Comment (igor-kamil) [reply public/css/less/bootstrap/glyphicons.less:1]: we don't use `glyphicons` at all. so I removed them - also from `resources/less/bootstrap/bootstrap.less`

File: .travis.yml [hunk 3]
```diff
@@ -1,8 +1,23 @@
 services:
   - docker
 
+cache:
+  directories:
+    - docker_images
+    - node_modules
+    - vendor  
+
```
[the remark this answers is not in the document]
Comment (igor-kamil) [reply .travis.yml:9]: you are right. also... it didn't help to speed it up at all :/

File: .travis.yml [hunk 4]
```diff
@@ -1,8 +1,23 @@
...
+  - nvm install 12
+  - docker load -i docker_images/images.tar || true
+
+before_script:  
   - cp .env.example .env
+  - npm install
+  - npm run testing
 
 script:
   - docker-compose run php vendor/bin/phpunit
```
[the remark this answers is not in the document]
Comment (igor-kamil) [reply .travis.yml:20]: well I see... but npm is not part of that container. it ends up with `exec: npm: not found`

File: Dockerfile [hunk 5]
```diff
@@ -9,13 +9,18 @@ RUN apt-get update -y && apt-get install -y \
...
 
+# Adding PHP extensions
 RUN docker-php-ext-configure gd --with-jpeg-dir=/usr/include/ \
     && docker-php-ext-install \
     pdo_mysql \
     gd \
     zip \
     exif
 
+# Install nodejs
```
Comment (eronisko) [inline Dockerfile:20]: I'd probably put this on line 5, e.g
```
RUN curl -sL https://deb.nodesource.com/setup_12.x | bash - \
  && apt-get update -y \
  && apt-get install -y \
  libpng-dev \
  ...
  nodejs
```

File: Dockerfile [hunk 6]
```diff
@@ -42,3 +47,5 @@ COPY . .
 
 # Re-run composer, this time with autoloader & scripts
 RUN composer install --optimize-autoloader
+
+RUN npm install
```
Comment (eronisko) [inline Dockerfile:51]: If you do something like this:
```
COPY package.json package-lock.json ./
RUN npm install
```
then it'll use cache unless either of those 2 files changes. If you put it after `COPY . .`, then it'll re-install everytime _any_ file is changed (which is most likely not what we want).

My suggestion would be to do the `npm install` on line 41 (after `WORKDIR /var/www`)

File: .travis.yml [hunk 7]
```diff
@@ -1,8 +1,8 @@
 services:
   - docker
 
-before_install:
+before_script:  
   - cp .env.example .env
 
 script:
-  - docker-compose run php vendor/bin/phpunit
+  - "docker-compose run php npm run testing && vendor/bin/phpunit"
```
Comment (eronisko) [inline .travis.yml:8]: maybe this'll work
```suggestion
  - docker-compose run php bash -c "npm run testing && vendor/bin/phpunit"
```

**Comment by eronisko** [comment]
I am so excited for this! 🔥

**Comment by igor-kamil** [comment]
thank you @eronisko for support.... finally it's fixed!

**Pull request merged**